File: src/kata11/nwiggins/testwordgame.rb
require 'test/unit'
require 'wordgame'

class Testwordgame < Test::Unit::TestCase

  def setup
    @game = WordGame.new()
  end

  def teardown
    @game = nil
  end

  def testCreateFourWordMap
    @game = WordGame.new()
    map = @game.createNodeMap("../fourletterwords.txt")
    assert map != nil
  end
  def testCreateFiveWordMap
    map = @game.createNodeMap("../fiveletterwords.txt")
    assert map != nil
  end
  def testSameFourWords
    path = @game.getPath('test', 'test')
    assert path.length == 1
  end
  def testSameFiveWords
    path = @game.getPath('place', 'place')
    assert path.length == 1
  end
  def testUnion
    test = @game.getAdjacentNodes("test", ['text', 'pest'], ['turn', 'text'])
    assert test.length == 2
  end
  def testGame
    test = @game.getPath('boat', 'toad')
    assert test.length == 4
    assert test == ['boat', 'goat', 'goad', 'toad']
  end
  def testLongGame
      test = @game.getPath('slams', 'sonly')
      assert test.length == 10
    end
end

File: src/kata11/nwiggins/testwordgame.py
'''
Created on Dec 20, 2010

@author: nwiggins
'''
import unittest

from wordgame import WordGame
class Test(unittest.TestCase):


    def setUp(self):
        self.game = WordGame()
        pass


    def tearDown(self):
        pass


    def testCreateFourWordMap(self):
        map = self.game.createNodeMap("../fourletterwords.txt")
        assert map != None
        pass
    def testCreateFiveWordMap(self):
        map = self.game.createNodeMap("../fiveletterwords.txt")
        assert map != None
        pass
    def testSameFourWords(self):
        path = self.game.getPath('test', 'test')
        assert len(path) == 1
        pass
    def testSameFiveWords(self):
        path = self.game.getPath('place', 'place')
        assert len(path) == 1
        pass
    def testUnion(self):
        test = self.game.getAdjacentNodes("test", ['text', 'pest'], ['turn', 'text'])
        assert len(test) == 2
        pass
    def testGame(self):
        test = self.game.getPath('boat', 'toad')
        assert len(test) == 4
        assert test == ['boat', 'goat', 'goad', 'toad']
        pass
    def testLongGame(self):
        test = self.game.getPath('slams', 'sonly')
        assert len(test) == 10
        pass
if __name__ == "__main__":
    #import sys;sys.argv = ['', 'Test.testName']
    unittest.main()